src/emailcontactproxymodel.h

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2020 - 2021 Open Mobile Platform LLC.
+ *
+ * This program is licensed under the terms and conditions of the
+ * Apache License, version 2.0.  The full text of the Apache License is at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+#ifndef EMAILCONTAXTPROXYMODEL_H
+#define EMAILCONTACTPROXYMODEL_H
+
+#include <QRegExp>
+#include <QSortFilterProxyModel>
+
+
+class Q_DECL_EXPORT EmailContactProxyModel : public QSortFilterProxyModel
+{
+    Q_OBJECT
+    Q_PROPERTY(QString filter READ filter WRITE setFilter NOTIFY filterChanged)
+    Q_PROPERTY(int count READ count NOTIFY countChanged)
+
+public:
+    explicit EmailContactProxyModel(QObject *parent = nullptr);
+    ~EmailContactProxyModel() override = default;
+
+    QHash<int, QByteArray> roleNames() const override;
+    bool filterAcceptsRow(int source_row, const QModelIndex &source_parent = QModelIndex()) const override;
+    virtual int count() const;
+    QString filter() const;
+
+    Q_INVOKABLE void setMaxCount(const int maxCount);
+
+public slots:
```

Comment:
No need to declare slot for filter if it's already a property. Better if the qml doesn't get a chance to call both filter = "foo" and setFilter("foo")

Reply:
done